raft/log.go

```diff
@@ -353,8 +353,7 @@ func (l *raftLog) mustCheckOutOfBounds(lo, hi uint64) error {
 		return ErrCompacted
 	}
 
-	length := l.lastIndex() + 1 - fi
-	if lo < fi || hi > fi+length {
+	if hi > l.lastIndex()+1 {
```

Comment:
I agree with you.
The original intention of my revision is because I think the test expression `lo < fi` is redundant. It has already return `ErrCompacted` in the above check. 

```
func (l *raftLog) mustCheckOutOfBounds(lo, hi uint64) error {
        ...

	if lo < fi {
		return ErrCompacted
	}
        
        ...

	if lo < fi || hi > fi+length {
		...
	}
        
        ...
}
```